### ISSUE TITLE
Allow type widening in no-unsafe-type-assertion

### ISSUE THREAD

**Issue by danielnixon** [danielnixon/eslint-plugin-total-functions#20]
Type assertions can be used to widen types. The no-unsafe-type-assertion rule currently flags those usages. It probably shouldn't.

E.g. these should pass the rule:
```typescript
const foo = "foo" as string;
const bar = 1 as number;
```

**Comment by danielnixon** [comment]
Once we have this let's add no-unsafe-type-assertion to the recommended config.
